### DISCUSSION TITLE
CtrlrDocs

### DISCUSSION THREAD

**Discussion by unityconstruct** [RomanKubiak/ctrlr#607] [General]
CtrlrDocs repo created & made private pending the process of consolidation & cleanup to avoid causing confusion & moot questions on this forum

**Comment by Godlike-Productions** [comment]
Feel free to grab my Wiki posts from CtrlrX.  I started to put snippets and stuff in there to help others (and to remind myself for when I hit the same hurdle later).

**Comment by dobo365** [comment]
Well... Thx for your effort...but...
When coming on this site, I'm not feeling I'm reaching some help pages. I (but this is my humble opinion) don't feel supported by the hierarchy.
Then in your example above I see that you copied extracts from the old forum posted by human fly.
That user joined at a certain moment then has been extremely verbose but was starting from scratch and was explaining all the steps and tests he has been doing. Taking source code and explanations from him should, in my opinion, be taken with caution.

As we see the changes and progress done by Damien with CtrlrX, I think it would be better to re-arrange its Wiki as I did my Step by Step guide 2.0 and continue there. I checked the Wiki a few weeks ago and was going to propose him some new structure.

I see it like this:
- Installing Ctrlr then building a simple ADSR control panel
- Building a more complex panel for an existing synth
- Automating panels with Lua programming
- Ctrlr source code - Compiling and building Ctrlr
- Ctrlr Reference

And I have sub chapters under those main headers

For further discussion ;-)

**Reply by Ted-juh** [reply]
I have to agree with Goodweather. Be very cautious when it comes to providing help with information from the forums or discussions page. It's good to get a general idea or get the big picture but when it comes to the code itself I would be very cautious indeed. There's just a few people whose code I trust. As a matter of fact, it depends on so many things if the code is suitable for your case. It's a guideline, not a "copy and paste and it will work" thing that you can do blindly.

I support giving straight outlined information about how to do certain things in a chronological order, the way Goodweather suggests, to get to a functional panel. And "straight outlined" could as well mean "well thought out" or "in an understandable way". 

And although I admire the enthusiasm and effort you've taken, the outcome is quite confusing. Maybe suitable for you but not the majority.

To add a chapter to the wiki Goodweather proposes, what about "Ctrlr's Source code structure explained" or "Bug fixing and Extending Ctrlr"? Just to not let Damien do all the hard work.

**Reply by Godlike-Productions** [reply]
Just to add to this, you also need to take into account the date of the code in the ctrlr forums.

Alot of the earlier code won't work any more as ctrlr changed a lot.

The example panels are a pretty good source of how to do a lot, and putting those into some support documents with additional explanation might be of benefit to new users.

**Comment by unityconstruct** [comment]
### TDLTR;
To the extent `CtrlrDocs/Panels/Lua repos` are a disruption to those here or on CtrlrX, I can set those Private while I sift through things, clean, trash & polish the data therein. I originally planned to go dark with it all while I go through it, but felt somewhat compelled to to leave it all in the same public domain in which I found it - particularly due to it representng years of blood, sweat & tears by someone other than myself. I also felt obligated to FYI those here, given they were intimatetly involved with the `Ctrlr Golden Years of Failure & Triumph`

<hr/>


lol.. Absolutely agree with all you are intimating brother!.. Speaking as one not benefiting from 10+ years on the scene, can definitely attest the confusion you observe is likely amplified for someone attempting a structured approach to understand what Ctrlr is, what's all there, then obviously how to use it without it develoving into an hourly Q&A session targeting the few active folks at every turn for lack of understanding fundamentals.
- Pursuent to that - and likely in my classic boisterous fashion - opted to simply collect everything on the planet that relates to Crtrl into one place so it can be parsed to even see what's there,  rather than keep sifting through endless/fruitless web searches hoping for the golden nuget of the day, & rather than spam the Discusssions with endless one-off queries. 
- To your point, going through those 60-blog-lessons it was readily apparent that while the 'right' information IS there, getting to & actually discerning it is likey an expensive & fruitless endeavor. Pulling everthing off Ctrlr.org has allowed me to quickly `Find in files`, parse the hits for some examples, and move on.
- The two PDFs for build & step-by-step are simply amazing productions that got me up to speed, but once up to speed there's little in the way of a 'Master Class' or more holistic representation of what Ctrlr is that would enable someone to be self-sufficient. There's really no references. In the same spirit, I'm necesarily in the position to basically repeat that process of distilling absurd amounts of niche information into some functional format - which may or may not be suitable for some, but definitely has been solving problems before they get started while decontructing a panel i've been using for years & wanted whose mechanics I need to understand.
- The repos I created were expressly stood up for purposes of this distillation - & the ensuing chaos stemming from it - to be conducted in a more discrete fashion as to not further burden the small community of SMEs that steadily shrinks over the years. Actually the generously provided information over the last two weeks has been highly instrumental in getting me along far enough where I'm self sufficient and if you've seen the sheer amount information collected thus far we probably agree I have plenty to keep me busy for the foreseeable future. 😭

I appreciate you flagging the concerns & let me know if you'd like those Repos set to `Private` brother.
#### Thanks again for all the info that got me over the initial learning curve!!!



<hr/>

Just FYI, regarding the challenges of anyone desiring to jump into Roman's WonderApp:  Resources are incredibly scattered & the original Ctrlr.org is largely defunct with no alternative for learning.


Two case in points giving rise to the need for me to do what I'm doing:
- I had archive.org kick off a site backup of Ctrlr.org, that has been running since 2-12 on the account of unrelenting spam likely in the 100's of thousands that obfuscate anything that is/was there.
![image](https://github.com/RomanKubiak/ctrlr/assets/40268430/3414125e-8033-482d-8efe-86598fa5da69)
![image](https://github.com/RomanKubiak/ctrlr/assets/40268430/00ddaed3-604a-4a8a-b9ed-b05739db076e)

- I spent no less than 4hrs yesterday chasing down the NRPN thing - a large part of which was spent parsing endless discussion that ultimately concluded ( after 14 pages of discussion) with dogmatic statements that really didn't answer the question & just reverted to brute force step through to reverse engineer the logic & which components are involved. If that answer is readily available online I'll fly out to take my lickin's 😎
![image](https://github.com/RomanKubiak/ctrlr/assets/40268430/20e2a1b7-07d4-4a2d-afff-6e0e5e1c8c69)

**Reply by dnaldoog** [reply]
Nirpins my nemesis - 😂
